### PR TITLE
[GitHub  #1182] add providers controller

### PR DESCRIPTION
**Pull request by sudhanshug16**
partially solves for #1182

### CODE REVIEW

**Review by khaliqgant — APPROVED** [review]
Works as expected. I'll add a follow up PR which will add documentation to this.

**Comment by sudhanshug16** [comment]
@khaliqgant can you please merge this at your convenience.

File: packages/server/lib/controllers/provider.controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class ProviderController {
+    /**
+     * Webapp
+     */
+
+    async listProviders(req: Request, res: Response, next: NextFunction) {
+        try {
+            let templates = configService.templates ?? {};
+            const query = req.query['query'] as string | undefined;
+            const limit = parseInt((req.query['limit'] as string) ?? '10');
```
Comment (khaliqgant) [inline packages/server/lib/controllers/provider.controller.ts]: @sudhanshug16 forgot to ask a question on this limit parameter? What was your use case here? It doesn’t seem very useful without a pagination interface

Comment (sudhanshug16) [reply packages/server/lib/controllers/provider.controller.ts]: I wanted to limit the size of the output when paired with `query`. I agree it doesn't make sense without a pagination interface.
Removed it.

Comment (sudhanshug16) [reply packages/server/lib/controllers/provider.controller.ts]: @khaliqgant, gentle bump.

Would really appreciate it if this was merged asap, my launch depends on it :)

Comment (khaliqgant) [reply packages/server/lib/controllers/provider.controller.ts]: No problem. Merging now and will deploy shortly!

Comment (khaliqgant) [reply packages/server/lib/controllers/provider.controller.ts]: This has been deployed @sudhanshug16

**Pull request merged**